### PR TITLE
add options for search input

### PR DESCRIPTION
**Pull request by madhur-tandon**
Closes https://github.com/jupyterlab-contrib/search-replace/issues/26

I think more work is needed in terms of UI + check if results are being changed on toggling of the options.

### CODE REVIEW

File: src/searchReplace.tsx [hunk 1]
```diff
@@ -29,14 +37,52 @@ export class SearchReplaceModel extends VDomModel {
...
+  }
+
+  get useRegex(): boolean {
+    return this._useRegex;
+  }
+
+  set useRegex(v: boolean) {
+    this._useRegex = v;
+    this.stateChanged.emit();
+  }
```
Comment (fcollonval) [inline src/searchReplace.tsx:65]: I would add the new search trigger in the setters. So the model will be consistent and in the view you don't need to call the search update.

> You should set `getSearchString` as private and never trigger it. But always invoke it through the debouncer.

**Comment by fcollonval** [comment]
Indeed you need to update the styles - I would align on the same row the search box and he buttons (you can try setting the `minimal` attribute on the buttons if they look too big).

Definitely need to add tests.

File: src/searchReplace.tsx (hunk 1 above)
Comment (madhur-tandon) [reply src/searchReplace.tsx]: done

**Comment by madhur-tandon** [comment]
@fcollonval should be ready to merge I guess

File: ui-tests/tests/search.spec.ts [hunk 2]
```diff
@@ -71,3 +71,28 @@ test('should get no matches', async ({ page }) => {
     await page.waitForSelector('#jp-search-replace >> text="No Matches Found"')
   ).toBeTruthy();
 });
+
+test('should test for case sensitive option', async ({ page }) => {
```
Comment (fcollonval) [inline ui-tests/tests/search.spec.ts:75]: Could you add test to the tow other buttons please?

**Review by fcollonval — CHANGES_REQUESTED** [review]
Styling could be improved. But mainly you should add test for the two other buttons.

File: ui-tests/tests/search.spec.ts (hunk 2 above)
Comment (madhur-tandon) [reply ui-tests/tests/search.spec.ts:75]: done

**Comment by madhur-tandon** [comment]
@fcollonval this is ready for merge!

**Review by fcollonval — APPROVED** [review]
LGTM

**Pull request merged**